extensions/amp-audio/1.0/amp-audio.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseElement} from './base-element';
+import {closestAncestorElementBySelector} from '#core/dom/query';
+import {dict} from '#core/types/object';
+import {isExperimentOn} from '#experiments';
+import {Layout} from '#core/dom/layout';
+import {triggerAnalyticsEvent} from '../../../src/analytics';
+import {user, userAssert} from '../../../src/log';
+import {validateMediaMetadata} from '../../../src/mediasession-helper';
+
+/** @const {string} */
+const TAG = 'amp-audio';
+const STORY_TAG = 'AMP-STORY';
+
+/**
+ * Check for element's action invocation validity
+ * @param {*} element Element to be check for validity
+ * @return {boolean} Returns true on valid invocation otherwise false
+ */
+function isInvocationValid(element) {
+  // Actions won't work with `<amp-story>` element
+  if (closestAncestorElementBySelector(element, STORY_TAG)) {
+    user().warn(
+      TAG,
+      '<amp-story> elements do not support actions on <amp-audio> elements'
+    );
+    return false;
+  }
+  return true;
+}
```

Comment:
I've confirmed that the check is not necessary.

@gmajoulet I'm removing the check from the Classic implementation on #35643.
@AnuragVasanwala Please remove this check from the Bento implementation.

Reply:
Thank you so much for the quick review.
Let me remove `isInvocationValid`  from the Bento implementation.

Reply:
@alanorozco

`audioPlaying` ([now `isPlaying`](https://github.com/ampproject/amphtml/pull/35342/commits/56b6b9e7fa4655cf392316924a0d777ad6871c5e)) will only be called when audio/video is going to be played. So, I can safely move `onPlay` into `isPlaying` event handler.

And for `onPause` prop, I need to add `onPause` attribute on `audio` element as:
```diff
/**
 * Plays audio callback
 */
const play = useCallback(() => {
- onPlay?.();
  audioRef.current.play();
  isPlaying.current = true;
}, []);
/**
 * Pauses audio callback
 */
const pause = useCallback(() => {
- onPause?.();
  audioRef.current.pause();
  isPlaying.current = false;
}, []);

/**
 * Updates media session for current window/tab
 */
const onPlaying = useCallback(() => {
...
+ onPlay?.();
...
}, [...]);

+ const onPaused = useCallback(() => {
+ ...
+ onPause?.();
+ ...
+ }, [...]);
...
<ContainWrapper contentRef={wrapperRef} size layout paint {...rest}>
  <audio
    ref={audioRef}
    aria-describedby={ariaDescribedby}
    aria-label={ariaLabel}
    aria-labelledby={ariaLabelledby}
    autoplay={autoplay}
    controls // Force controls otherwise there is no player UI.
    controlsList={controlsList}
    loading={loading}
    loop={loop}
    muted={muted}
    onPlaying={() => onPlaying()}
+   onPause={() => onPaused()}
    preload={preload}
    src={src}
  >
    {sources}
  </audio>
</ContainWrapper>

```

I think this is what you are suggesting.
Please let me know if I've missed anything.